server/__tests__/snake.test.js

```diff
@@ -62,18 +61,135 @@ describe('Test Snake Class', () => {
   describe('method getFoodReward', () => {
     it('should give food reward to corresponding snake', () => {
       snake.eatFood = () => true;
-      const randomhead = { x: 234, y: 982 };
-      snake.head = randomhead;
+      const bodyArrayLengthOfSnakeBeforeEating = snake.bodyCoordinates.length;
       const snakeLength = snake.length;
       const lifeLine = snake.life;
       snake.getFoodReward();
-      expect(randomhead).toEqual(snake.head);
-      expect(randomhead).toEqual(snake.bodyCoordinates[snake.length - 1]);
-      expect(snakeLength + SNAKE.FOOD_LENGTH_REWARD).toEqual(snake.length);
-      expect(snakeLength + SNAKE.FOOD_LENGTH_REWARD).toEqual(snake.bodyCoordinates.length);
-      expect(lifeLine + SNAKE.FOOD_LIFE_REWARD).toEqual(snake.life);
+      expect(snakeLength + SNAKE_CONSTANTS.FOOD_LENGTH_REWARD).toEqual(snake.length);
+      expect(bodyArrayLengthOfSnakeBeforeEating).toEqual(snake.bodyCoordinates.length);
+      expect(lifeLine + SNAKE_CONSTANTS.FOOD_LIFE_REWARD).toEqual(snake.life);
     });
+  });
 
+  describe('method hasCollidedWithItself', () => {
+    it('should return true for this method call after setting given below coordinates', () => {
```

Comment:
Something like `it('checks snake's collisions with itself')` would make more sense than just telling what the function returns.

Reply:
Yes, that's sounds good.